tests/test_engine.py

```diff
@@ -2524,3 +2674,31 @@ async def mock_resolve_inputs(*args, **kwargs):
             )
 
     assert "Task parameter introspection took" not in caplog.text
+
+
+async def test__check_task_failure_retriable_not_callable(
+    flow_run, prefect_client, caplog: pytest.LogCaptureFixture
+):
+    @task(retry_condition_fn="not a callable")
+    def my_task():
```

Comment:
After moving the `retry_condition_fn` validation to `tasks.py` this could go in `test_tasks.py`